### PR TITLE
Java: Update MaD Declarations after Triage

### PR DESCRIPTION
**Pull request by kaeluka**
This PR updates the MaD declarations after a recent triage session.

TR 447

### CODE REVIEW

File: java/ql/lib/ext/java.awt.model.yml
Comment (kaeluka) [inline java/ql/lib/ext/java.awt.model.yml:20]: this might be a misclassification, I'm not sure whether this might actually be a url-redirection (or something new altogether)

Comment (atorralba) [reply java/ql/lib/ext/java.awt.model.yml:20]: Hard to tell.

On one hand, `url-redirection` is all about redirecting the user's browser, which is a separate component from the vulnerable application (the webapp running in a server), and in this case you're making a redirect (i.e. forcing a request) in the same machine where the vulnerable application lives, which sounds more like `request-forgery`.

But on the other hand, the fact that it redirects a _browser_, and doesn't really perform a server-to-server request makes it sound more like `url-redirection`.

I think neither match 1:1 in a desktop application, but if I had to choose I think I'd be slightly inclined to label it as `url-redirection` instead.

```suggestion
      - ["java.awt", "Desktop", True, "browse", "(URI)", "", "Argument[0]", "url-redirection", "ai-manual"]
```

Also, I don't expect many Java AWT applications to be left out there, so this probably won't have much impact :)

File: java/ql/lib/ext/io.undertow.server.handlers.resource.model.yml
Comment (atorralba) [inline java/ql/lib/ext/io.undertow.server.handlers.resource.model.yml:6]: Judging by the methods in the [documentation](https://undertow.io/javadoc/2.0.x/io/undertow/server/handlers/resource/URLResource.html) and some [code search](https://github.com/search?type=code&q=lang%3Ajava+%22import+io.undertow.server.handlers.resource.URLResource%3B%22), this looks like a class you use to expose (i.e. serve) some resource, not to request it. So it shouldn't qualify as `request-forgery`.

File: java/ql/lib/ext/javax.servlet.http.model.yml
Comment (atorralba) [inline java/ql/lib/ext/javax.servlet.http.model.yml:29]: This is already handled in regular CodeQL: https://github.com/github/codeql/blob/67f8bcce440017cb8bd63656538c4fcec00d3202/java/ql/lib/semmle/code/java/security/UrlRedirect.qll#L19

I'm not sure what to do in these cases. We could merge it and accept the duplicate until we clean up, merge it only if we also clean it up in the same PR, or not merge it at all.

Comment (kaeluka) [reply java/ql/lib/ext/javax.servlet.http.model.yml]: cd765e7 I've commented the model out, which is what we've done previously in similar situations

File: java/ql/lib/ext/io.undertow.server.handlers.resource.model.yml
Comment (kaeluka) [reply java/ql/lib/ext/io.undertow.server.handlers.resource.model.yml]: cd765e7 thank you, I've removed the model!

File: java/ql/lib/ext/java.awt.model.yml
Comment (kaeluka) [reply java/ql/lib/ext/java.awt.model.yml]: Thanks, I've applied your suggestion.

> Also, I don't expect many Java AWT applications to be left out there, so this probably won't have much impact :)

["Write once, run anywhere!"](https://en.wikipedia.org/wiki/Write_once,_run_anywhere) ;)

File: java/ql/lib/change-notes/2023-12-21-new-models.md
Comment (atorralba) [inline java/ql/lib/change-notes/2023-12-21-new-models.md:8]:
```suggestion
* Added a dataflow model for `java.awt.Desktop.browse(URI)`.
```

**Review by atorralba — APPROVED** [review]
Thanks for addressing the comments!

**Pull request merged**
